R/tspa_plot.R

```diff
@@ -0,0 +1,269 @@
+#' Diagnostic plots of fitted 2S-PA model
+#'
+#' @param tspa_fit An object of class \code{lavaan},
+#'                 representing the output generated from `tspa()` function.
+#' @param title Character. Set the name of scatter plot. The default value is "Scatterplot".
+#' @param label_x Character. Set the name of the x-axis. The default value is "fs_" followed by variable names.
+#' @param label_y Character. Set the name of the y-axis. The default value is "fs_" followed by variable names.
+#' @param ... Additional arguments passed to \code{\link[graphics]{plot}}. See
+#'            \code{\link[graphics]{plot}} for a list.
+#' @param abbreviation Logic input. If 'FALSE' is indicated, the group name will be shown in full.
+#'                     The default setting is 'True'.
+#' @param ask Logic input. If 'TRUE' is indicated, the user will be asked before
+#'            before each plot is generated. The default setting is 'False'.
+#' @param fscore_type Character. Set the type of factor score for input.
+#'                    The default setting is using factor score from observed data (i.e., output from 'get_fs').
+#'                    If 'fscore_type = "est"', then it will use output from 'lavPredict'
+#' @param ... Additional arguments passed to \code{\link[graphics]{plot}}. See
+#'            \code{\link[graphics]{plot}} for a list.
+#'
+#' @return A scatterplot between factor scores, and a residual plot.
+#'
+#' @export
+#'
+#' @examples
+#' library(lavaan)
+#' model <- '
+#' # latent variable definitions
+#' ind60 =~ x1 + x2 + x3
+#' dem60 =~ y1 + a*y2 + b*y3 + c*y4
+#' # regressions
+#' dem60 ~ ind60
+#' '
+#' fs_dat_ind60 <- get_fs(data = PoliticalDemocracy,
+#'                        model = "ind60 =~ x1 + x2 + x3")
+#' fs_dat_dem60 <- get_fs(data = PoliticalDemocracy,
+#'                        model = "dem60 =~ y1 + y2 + y3 + y4")
+#' fs_dat <- cbind(fs_dat_ind60, fs_dat_dem60)
+#'
+#' tspa_fit <- tspa(model = "dem60 ~ ind60",
+#'                  data = fs_dat,
+#' se = list(ind60 = 0.1213615, dem60 = 0.6756472))
+#' tspa_plot(tspa_fit)
+
+# Helper function for the scatter plot
+# Comment: put it outside the main function
+plot_scatter <- function (fscores_df, iv, dv,
```

Comment:
Right now the documentation applies to `plot_scatter()`. Please keep the documentation close to the main function, and move helpers to the bottom.